master/internal/resourcemanagers/fitting_methods.go

```diff
@@ -12,20 +12,62 @@ func labelSatisfied(req *AllocateRequest, agent *agentState) bool {
 	return req.Label == agent.label
 }
 
+func maxZeroSlotContainersSatisfied(req *AllocateRequest, agent *agentState) bool {
+	if req.SlotsNeeded == 0 {
+		if agent.maxZeroSlotContainers == nil {
+			return true
+		}
+		if *agent.maxZeroSlotContainers == 0 {
+			return false
+		}
+		return agent.numZeroSlotContainers() < *agent.maxZeroSlotContainers
+	}
+	return true
+}
+
+func agentIdleSatisfied(_ *AllocateRequest, agent *agentState) bool {
+	return agent.idle()
+}
+
 // Soft Constraints
 
 // BestFit returns a float affinity score between 0 and 1 for the affinity between the task and
 // the agent. This method attempts to allocate tasks to the agent that is both most utilized and
 // offers the fewest slots. This method should be used when the cluster is dominated by multi-slot
 // applications.
-func BestFit(_ *AllocateRequest, agent *agentState) float64 {
+func BestFit(req *AllocateRequest, agent *agentState) float64 {
+	if req.SlotsNeeded == 0 {
+		if agent.maxZeroSlotContainers == nil {
+			if agent.numZeroSlotContainers() == 0 {
+				return 0
+			}
+			return 1.0
+		}
+		if *agent.maxZeroSlotContainers == 0 {
+			return 0.0
+		}
+		return 1.0 / (1.0 + float64(*agent.maxZeroSlotContainers-agent.numZeroSlotContainers()))
+	}
 	return 1.0 / (1.0 + float64(agent.numEmptySlots()))
 }
 
 // WorstFit returns a float affinity score between 0 and 1 for the affinity between the task and
 // the agent. This method attempts to allocate tasks to the agent that is least utilized. This
 // method should be used when the cluster is dominated by single-slot applications.
-func WorstFit(_ *AllocateRequest, agent *agentState) float64 {
+func WorstFit(req *AllocateRequest, agent *agentState) float64 {
+	if req.SlotsNeeded == 0 {
+		if agent.maxZeroSlotContainers == nil {
+			if agent.numZeroSlotContainers() == 0 {
+				return 1.0
+			}
+			return 0
```

Comment:
Shouldn't the affinity be impacted by how many zero slot task are running on the agent?